core/trino-main/src/test/java/io/trino/operator/PageTestUtils.java

```diff
@@ -75,24 +75,35 @@ public static Page createRandomPage(
             float nullRate,
             Optional<Wrapping> wrapping)
     {
-        int channelCount = types.size();
-        ImmutableList.Builder<Block> blocks = ImmutableList.builder();
+        List<Block> blocks = types.stream()
+                .map(type -> createRandomBlock(type, positionCount, nullRate, wrapping))
+                .collect(toImmutableList());
 
-        for (int i = 0; i < channelCount; i++) {
-            Block block = createRandomBlockForType(types.get(i), positionCount, nullRate);
-            blocks.add(wrapping.map(w -> w.wrap(block, positionCount)).orElse(block));
-        }
+        return createRandomPage(types, positionCount, hashChannels, blocks);
+    }
 
-        hashChannels.ifPresent(channels -> {
-            ImmutableList<Block> blocksWithoutHash = blocks.build();
+    private static Block createRandomBlock(Type type, int positionCount, float nullRate, Optional<Wrapping> wrapping)
+    {
+        Block block = createRandomBlockForType(type, positionCount, nullRate);
+        return wrapping.map(w -> w.wrap(block, positionCount)).orElse(block);
+    }
 
-            blocks.add(getHashBlock(
+    public static Page createRandomPage(
```

Comment:
How is this method creating random page if it get blocks as input? There is nothing random about this method. Consider some different name

Reply:
good catch. changed it just `createPage` but I'm also considering `createPageWithHashBlock` WDYT?